connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala

```diff
@@ -79,7 +105,11 @@ private[sql] class SparkConnectClient(
       .setSessionId(sessionId)
       .setClientType(userAgent)
       .build()
-    stub.executePlan(request)
+    retry {
+      val result = stub.executePlan(request)
+      result.hasNext // moves evaluation of BlockingResponseStream to SparkConnectClient
```

Comment:
I think you should create an iterator that wraps this (?). e.g., does it cover if it fails in the middle of the next iteration?

Reply:
Good catch! Added a wrapper for this that also removes this ugly workaround.